openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```diff
@@ -0,0 +1,133 @@
+package managementcpusoverride
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"k8s.io/apiserver/pkg/admission"
+	coreapi "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/kubelet/cm"
+)
+
+const (
+	PluginName                 = "autoscaling.openshift.io/ManagementCPUsOverride"
+	podWorkloadManagementLabel = "workload.openshift.io/target=management"
+	containerCPUsAnnotation    = "io.openshift.workload.management.cpu"
+)
+
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName,
+		func(config io.Reader) (admission.Interface, error) {
+			return &managementCPUsOverride{
+				Handler: admission.NewHandler(admission.Create),
+			}, nil
+		})
+}
+
+type managementCPUsOverride struct {
+	*admission.Handler
+}
+
+var _ = admission.MutationInterface(&managementCPUsOverride{})
+
+func (a *managementCPUsOverride) Admit(ctx context.Context, attr admission.Attributes, o admission.ObjectInterfaces) error {
+	if attr.GetResource().GroupResource() != coreapi.Resource("pods") || attr.GetSubresource() != "" {
+		return nil
+	}
+
+	pod, ok := attr.GetObject().(*coreapi.Pod)
+	if !ok {
+		return admission.NewForbidden(attr, fmt.Errorf("unexpected object: %#v", attr.GetObject()))
+	}
+
+	// TODO: probably we will need to refactor it once we will have more workload types supported
+	if _, ok := pod.Labels[podWorkloadManagementLabel]; !ok {
+		return nil
+	}
+
+	// the only relevant pod QoS class that we interested is Burstable one
+	// only relevant resources to decide the QoS class is CPU and memory
+	if !isBurstable(pod, []coreapi.ResourceName{coreapi.ResourceCPU, coreapi.ResourceMemory}) {
```

Comment:
I don't know enough to know if this is correct. Someone from scheduling/kubelet should indicate whether the logic is correct.